elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/QueryDsl.scala

```diff
@@ -72,6 +72,24 @@ trait QueryDsl extends DslCommons with SortDsl {
       new QueryRoot(query, fromOpt, sizeOpt, sorts, timeout, sourceFilter)
     }
   }
+  
+  // This api should be incorporated into QueryRoot in 2.0 when breaking changes are allowed.
+  // Before that, this api can serve to extend QueryRoot with backward compatibility.
+  class ExtendedQueryRoot(override val query: Query,
```

Comment:
oh right! Forgot about that bit of nastiness. If you want, you could add another optional parameter to the apply method. If `terminateAfter` was defined, we'd return an `ExtendedQueryRoot` instead. That way we hide the problem inside of the library. What do you think?